compiler/rustc_middle/src/hir/map/mod.rs

```diff
@@ -1310,3 +1306,64 @@ pub(super) fn hir_module_items(tcx: TyCtxt<'_>, module_id: LocalDefId) -> Module
         }
     }
 }
+
+pub(crate) fn hir_crate_items(tcx: TyCtxt<'_>, _: ()) -> ModuleItems {
+    let mut collector = CrateCollector {
+        tcx,
+        submodules: Vec::default(),
+        items: Vec::default(),
+        trait_items: Vec::default(),
+        impl_items: Vec::default(),
+        foreign_items: Vec::default(),
+    };
+
+    tcx.hir().walk_toplevel_module(&mut collector);
+
+    let CrateCollector { submodules, items, trait_items, impl_items, foreign_items, .. } =
+        collector;
+
+    return ModuleItems {
+        submodules: submodules.into_boxed_slice(),
+        items: items.into_boxed_slice(),
```

Comment:
Reading the code, I'm not sure that `items` starts with `CRATE_DEF_ID`.  It should.

Reply:
Now that you mention it. Looking at where the walk begins in `walk_toplevel_module`, it seems  like `CRATE_DEF_ID` does not get added to `items`. Am I reading that right?